docs/tutorial/dark-mode.md

```diff
@@ -0,0 +1,183 @@
+use Electron `>=7.0.0`. Both [Electron Packager][electron-packager] and
+[Electron Forge][electron-forge] have a
+[`darwinDarkModeSupport` option][packager-darwindarkmode-api]
+to automate the `Info.plist` changes during app build time.
+
+If you wish to opt-out while using Electron &gt; 8.0.0, you must
+set the `NSRequiresAquaSystemAppearance` key in the `Info.plist` file to
+`true`. Please note that Electron 8.0.0 and above will not let you opt-out
+of this theming, due to the use of the macOS 10.14 SDK.
+
+## Automatically update the native interfaces
+
+"Native interfaces" include the file picker, window border, dialogs, context
+menus, and more - anything where the UI comes from your operating system and
+not from your app. The default behavior is to opt into this automatic theming
+from the OS.
+
+## Automatically update your own interfaces
+
+If your app has its own dark mode, you should toggle it on and off in sync with
+the system's dark mode setting. You can do this by using the
+[prefer-color-scheme] CSS @media query.
+
+## Manually update your own interfaces
+
+If you want to manually switch between light/dark modes, you can do this by
+setting the desired mode in the
+[themeSource](https://www.electronjs.org/docs/api/native-theme#nativethemethemesource)
+property of the `nativeTheme` module.
+
+## Example
+
+We'll start with a working application from the
+[Quick Start Guide](quick-start.md) and add functionality gradually.
+
+First, let's add a page where a user can toggle between light and dark modes.
+By default, Electron will follow the system's dark mode preference, so we
+initially set the theme source as "System".
+
+Add the following lines to the `index.html` file:
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8">
+    <title>Hello World!</title>
+    <meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline';" />
+    <link rel="stylesheet" type="text/css" href="./styles.css">
+</head>
+<body>
+    <h1>Hello World!</h1>
+    <p>Current theme source: <strong id="theme-source">System</strong></p>
+
+    <button id="toggle-dark-mode">Toggle Dark Mode</button>
+    <button id="reset-to-system">Reset to System Theme</button>
+
+    <script src="renderer.js"></script>
+  </body>
+</body>
+</html>
+```
+
+Next, add event listeners to the toggle buttons and response handlers. The
+listeners will listen for click events and update the content on the web page
+accordingly:
+
+* when the "Toggle Dark Mode" button is clicked, Electron sends the
+`dark-mode:toggle` message (event) to the Main process that triggers theme
+change, and sets the Current theme source based on the response from the
+Main process.
+* when the "Reset to System Theme" button is clicked, Electron sends
+the `dark-mode:system` message (event) to the Main process that tells that
+the theme should be set to default and sets the Current theme source to
+`System`.
```

Comment:
```suggestion
* When the "Toggle Dark Mode" button is clicked, we send the
`dark-mode:toggle` message (event) to tell the Main process to trigger
a theme change, and update the "Current Theme Source" label in the UI based on the
response from the Main process.
* When the "Reset to System Theme" button is clicked, we send
the `dark-mode:system` message (event) to tell the Main process to use
the system color scheme, and update the "Current Theme Source" label
to `System`.
```

Some copy-editing suggestions here:
* Used `update the "Current Theme Source" label` rather than `sets the Current theme source` to avoid any ambiguity about what is being set in the renderer.
* Reworded a bit to make it clearer that we're just telling the Main process that we want to do stuff, not that any of the actions actually happen yet.